app/controllers/parties_controller.rb

```diff
@@ -0,0 +1,98 @@
+require "google/apis/calendar_v3"
+require "google/api_client/client_secrets.rb"
+
+class PartiesController < ApplicationController
+  CALENDAR_ID = 'primary'
+
+  def new
+    movie_data = MovieData.all_data(params[:movie_id])
+    @movie = MovieResult.new(movie_data)
+  end
+
+  def create
+    client = get_google_calendar_client(current_user)
+    viewing_party = current_user.parties.create(
+      movie_title: params[:movie_title],
+      duration_of_party: params[:duration],
+      friend_ids: params[:friend_ids],
+      date: params[:date],
+      time: params[:time]
+    )
+    event = get_viewing_party(viewing_party)
+    client.insert_event('primary', event)
+    flash[:notice] = 'Viewing Party was successfully added to calendar.'
+    redirect_to '/dashboard'
+  end
+
+  def get_google_calendar_client(current_user)
+    client = Google::Apis::CalendarV3::CalendarService.new
+    return unless (current_user.present? && current_user.token.present? && current_user.refresh_token.present?)
+    secrets = Google::APIClient::ClientSecrets.new({
+      "web" => {
+        "access_token" => current_user.token,
+        "refresh_token" => current_user.refresh_token,
+        "client_id" => ENV["GOOGLE_API_KEY"],
+        "client_secret" => ENV["GOOGLE_API_SECRET"]
+      }
+    })
+    begin
+      client.authorization = secrets.to_authorization
+      client.authorization.grant_type = "refresh_token"
+
+      if !current_user.present?
+        client.authorization.refresh!
+        current_user.update_attributes(
+          access_token: client.authorization.token,
+          refresh_token: client.authorization.refresh_token,
+          expires_at: client.authorization.expires_at.to_i
+        )
+      end
+    rescue => e
+      flash[:error] = 'Your token has been expired. Please login again with google.'
+      redirect_to :back
+    end
+    client
+  end
+
+  private
+
+  def get_viewing_party(viewing_party)
+    unless viewing_party[:friend_ids].nil?
+      attendees = viewing_party[:friend_ids].map do |id|
```

Comment:
Tricky that attendees had to be in this hash format! worked great once we figured that out.